x-pack/plugins/enterprise_search/public/applications/app_search/components/result_settings/sample_response/sample_response_logic.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { i18n } from '@kbn/i18n';
+
+import { flashAPIErrors } from '../../../../shared/flash_messages';
+
+import { HttpLogic } from '../../../../shared/http';
+import { EngineLogic } from '../../engine';
+
+import { SampleSearchResponse, ServerFieldResultSettingObject } from '../types';
+
+const NO_RESULTS_MESSAGE = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.resultSettings.sampleResponse.noResultsMessage',
+  { defaultMessage: 'No results.' }
+);
+
+const ERROR_MESSAGE = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.resultSettings.sampleResponse.errorMessage',
+  { defaultMessage: 'An error occured.' }
+);
+
+interface SampleResponseValues {
+  query: string;
+  response: SampleSearchResponse | string | null;
+}
+
+interface SampleResponseActions {
+  queryChanged: (query: string) => { query: string };
+  getSearchResultsSuccess: (response?: SampleSearchResponse) => { response?: SampleSearchResponse };
+  getSearchResultsFailure: (response?: string) => { response?: string };
+  getSearchResults: (
+    query: string,
+    resultFields: ServerFieldResultSettingObject
+  ) => { query: string; resultFields: ServerFieldResultSettingObject };
+}
+
+export const SampleResponseLogic = kea<MakeLogicType<SampleResponseValues, SampleResponseActions>>({
+  path: ['enterprise_search', 'app_search', 'sample_response_logic'],
+  actions: {
+    queryChanged: (query) => ({ query }),
+    getSearchResultsSuccess: (response) => ({ response }),
+    getSearchResultsFailure: (response) => ({ response }),
+    getSearchResults: (query, resultFields) => ({ query, resultFields }),
+  },
+  reducers: {
+    query: ['', { queryChanged: (_, { query }) => query }],
+    response: [
+      null,
+      {
+        getSearchResultsSuccess: (_, { response }) => response || null,
+        getSearchResultsFailure: (_, { response }) => response || null,
+      },
+    ],
+  },
+  listeners: ({ actions }) => ({
+    getSearchResults: async ({ query, resultFields }, breakpoint) => {
+      if (Object.keys(resultFields).length < 1) return;
+      await breakpoint(250);
+
+      const { http } = HttpLogic.values;
+      const { engineName } = EngineLogic.values;
+
+      const url = `/api/app_search/engines/${engineName}/sample_response_search`;
+
+      try {
+        const response = await http.post(url, {
+          body: JSON.stringify({
+            query,
+            result_fields: resultFields,
+          }),
+        });
+
+        const result = response.results?.[0];
+        actions.getSearchResultsSuccess(
+          result ? { ...result, _meta: undefined } : NO_RESULTS_MESSAGE
```

Comment:
setting `undefined` is a way cool one-liner alternative to `delete`ing the meta key! 💯

Reply:
Be careful, you can get into trouble with this

![Screen Shot 2021-04-02 at 3 19 23 PM](https://user-images.githubusercontent.com/2479295/113446923-d9b11180-93c6-11eb-95f0-882f0ae20bec.png)

setting a prop to undefined _is_ different than removing it entirely from an object

![Screen Shot 2021-04-02 at 3 20 39 PM](https://user-images.githubusercontent.com/2479295/113447021-0cf3a080-93c7-11eb-9b80-514749635799.png)

Reply:
Good point! I think it's OK for this use case since the output gets `JSON.stringify`ed which correctly eliminated undefined key/values, but ya it'd be an issue if we were checking key length.

Reply:
That's true Byron. This is actually how it was in the previous code base, just ported it over.

I probably would have written it like this:

```
const { _meta, ...updatedResult }  = result;
actions.getSearchResultsSuccess( result ? updatedResult : NO_RESULTS_MESSAGE
```